core/modules/wpublish/qserv-worker-status.cc

```diff
@@ -0,0 +1,210 @@
+// System header
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+// Third party headers
+#include "XrdSsi/XrdSsiProvider.hh"
+#include "XrdSsi/XrdSsiService.hh"
+
+// Qserv headers
+#include "global/ResourceUnit.h"
+#include "proto/worker.pb.h"
+#include "util/BlockPost.h"
+#include "util/CmdLineParser.h"
+#include "wpublish/GetStatusQservRequest.h"
+
+/// This C++ symbol is provided by the SSI shared library
+extern XrdSsiProvider* XrdSsiProviderClient;
+
+namespace global   = lsst::qserv;
+namespace util     = lsst::qserv::util;
+namespace wpublish = lsst::qserv::wpublish;
+
+using namespace std;
+
+namespace {
+
+// Command line parameters
+
+string workersFileName;
+unsigned int numRequests;
+string serviceProviderLocation;
+unsigned int numWorkers;
+bool workerFirst;
+unsigned int cancelAfterMs;
+
+vector<string> workers;
+
+bool readWorkersFile() {
+    ifstream file(workersFileName);
+    if (not file.good()) {
+        cerr << "error: failed to open a file with worker identifiers: "
+             << workersFileName << endl;
+        return false;
+    }
+    string worker;
+    while (file >> worker)
+        workers.push_back(worker);
+
+    if (not workers.size()) {
+        cerr << "error: no workers found in file with worker identifiers: "
+             << workersFileName << endl;
+        return false;
+    }
+    return true;
```

Comment:
readWorkersFile looks just like readWorkersFile() in qserv-worker-perf.cc and a few others. Perhaps a class that takes a file name and and holds a vector of strings read from the file?

Reply:
Good suggestion! I will try to eliminate this duplication!